test/e2e/framework/specs.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"github.com/onsi/ginkgo"
+)
+
+type spec struct {
+	text string
+	body func()
+}
+
+var specs []spec
+
+// DescribeWithTestContext is like ginkgo.Describe, but in contrast to
```

Comment:
I would like the comment to mention the motivation for why someone might want to use this function to "invoke spec bod[ies]" only after "TestContext has been fully initialized". i.e. this exists so that we can skip tests in spec bodies based on TestContext. Otherwise the file is too generic looking and the technical language imposing :]

Reply:
I understand "can add or parameterize specs depending on the test configuration" basically explains it but I would word it differently is all.

Reply:
I had another use case in mind: testing CSI drivers with the upstream e2e test binary.

With this change, the e2e test binary could have a command line parameter(s) that enable testing of a driver by providing the necessary information. The "TestDriver" instance that is given to the storage test suite then gets defined based on these parameters.

Without this change, the same can still be done, but would have to be based on parsing environment variables.